### DISCUSSION TITLE
What does chakra-cli actually do?

### DISCUSSION THREAD

**Discussion by frontendwizard** [chakra-ui/chakra-ui#3566] [Q&A]
I just came across it on the advanced theming section and having typings for custom themes is something I really want. I did the setup and run the command, and it gives me a success message but I don't get those types anywhere. So, I think we could improve the section explaining exactly what this command do, where do the types go and this kind of thing. My guess is that it mutates the types directly on node_modules, from the suggestion of adding a postinstall hook. This kind of information would be useful when debugging problems with the type generation.

**Comment by keithburgie** [comment]
@frontendwizard It adds your custom theme typings to a file at `../node_modules/@chakra-ui/styled-system/dist/theming.types.d.ts`

Now, any chance you could tell me about `postinstall`? 😅 Is this a command I should be familiar with? What about `gen:theme-typings`? I'm confused about the docs including this command, since it does the same thing as the "theme" script. Couldn't `postinstall` just run "theme"?

```
"scripts": {
   "theme": "chakra-cli tokens path/to/theme.ts",
   "theme:watch": "chakra-cli tokens path/to/theme.ts --watch",
  "gen:theme-typings": "chakra-cli tokens <path/to/your/theme.(js|ts)>",
  "postinstall": "npm run gen:theme-typings"
},
```
